### ISSUE TITLE
Plugin setAttributes error due to getting/setting of settings behaviour

### ISSUE THREAD

**Issue by joshuabaker** [craftcms/cms#4706]
### Description

`Plugin::setSettings` is called where a plugin is initiated, however, the default behaviour is to call `Plugin::getSettings`, which can return `null`. This results in exceptions for plugins that do not have settings.

```
Error: Call to a member function setAttributes() on null
```

### Reference

https://github.com/craftcms/cms/blob/1239460455a61552510153fd249e48e3c325c27f/src/base/Plugin.php#L195-L206

https://github.com/craftcms/cms/blob/1239460455a61552510153fd249e48e3c325c27f/src/base/Plugin.php#L211-L214

**Comment by joshuabaker** [comment]
Related issue: craftcms/postmark#10

**Comment by brandonkelly** [comment]
This looks like a Postmark bug so I’ll fix it over there, but in the meantime you should be able to fix this locally by clearing out your `settings` column value for the `postmark` plugin in your `plugins` table.

**Comment by brandonkelly** [comment]
While this was really a Postmark bug, it turns out that there’s no way for the plugin to actually fix the underlying issue without an update on Craft’s end to be more graceful if `setSettings()` is called for a plugin without settings. So I ended up implementing a fix similar to #4707 for the next Craft release.

To get the fix early, change your `craftcms/cms` requirement in composer.json to:

```json
"require": {
  "craftcms/cms": "dev-develop#36ed8fa14d27670a5ff7293886ffa8ccff5cc5e7 as 3.2.9",
  "...": "..."
}
```

Then run `composer update`.

Once you’ve done that, you should update to Postmark 2.0.2, which fixes the underlying issue. Until you do that, Craft will start logging warnings to `storage/logs/web.log` on every request.
